src/blocks/memory.rs

```diff
@@ -271,12 +313,47 @@ impl Memstate {
                 "SwapTotal:" => mem_state.swap_total = val,
                 "SwapFree:" => mem_state.swap_free = val,
                 "SwapCached:" => mem_state.swap_cached = val,
+                "Zswap:" => mem_state.zswap_compressed = val,
+                "Zswapped:" => mem_state.zswap_decompressed = val,
                 _ => (),
             }
 
             line.clear();
         }
 
+        // For ZRAM
+        if let Ok(entries) = std::fs::read_dir("/sys/block/") {
```

Comment:
This can be done with tokio like this:

```rust
        let mut entries = read_dir("/sys/block/")
            .await
            .error("Could not read /sys/block")?;
        while let Some(entry) = entries
            .next_entry()
            .await
            .error("Could not get next file /sys/block")?
        {
            if let Some(name) = entry.file_name().to_str() {
                if name.starts_with("zram") {
                    let Ok(file) = File::open(entry.path()).await else {
                        break;
                    };
                    ...
```

 I don't think it matters for `/sys` and `/dev`
 
 Not being able to read from `/sys` would be an error that should be raised.

Reply:
Whoops, good catch!